### PR TITLE
EE-1050: remove system contract wasms

### PR DESCRIPTION
**Pull request by EdHastingsCasperLabs**
This PR removes the wasm variants of system contracts (mint, auction, proof of stake, standard payment) and all related logic.

https://casperlabs.atlassian.net/browse/EE-1050

### CODE REVIEW

**Review by dwerner — APPROVED** [review]
Had a read over everything and didn't have any nitpicks/noteworthy suggestions. lgtm

File: execution_engine/src/core/engine_state/genesis.rs [hunk 1]
```diff
@@ -13,17 +14,61 @@ use casper_types::{
...
+        METHOD_BALANCE, METHOD_CREATE, METHOD_MINT, METHOD_READ_BASE_ROUND_REWARD,
+        METHOD_REDUCE_TOTAL_SUPPLY, METHOD_TRANSFER, ROUND_SEIGNIORAGE_RATE_KEY, TOTAL_SUPPLY_KEY,
+    },
+    proof_of_stake::{
+        ARG_ACCOUNT, METHOD_FINALIZE_PAYMENT, METHOD_GET_PAYMENT_PURSE, METHOD_GET_REFUND_PURSE,
+        METHOD_SET_REFUND_PURSE,
+    },
+    standard_payment::METHOD_PAY,
+    system_contract_errors::mint,
 };
```
Comment (henrytill) [inline execution_engine/src/core/engine_state/genesis.rs:72]: This should go above the crate-local imports.

File: execution_engine/src/core/engine_state/genesis.rs [hunk 2]
```diff
@@ -417,6 +418,831 @@ impl Distribution<ExecConfig> for Standard {
...
+            total_supply_uref
+        };
+
+        let named_keys = {
+            let mut named_keys = NamedKeys::new();
+            named_keys.insert(
+                ROUND_SEIGNIORAGE_RATE_KEY.to_string(),
+                round_seigniorage_rate_uref.into(),
+            );
+
```
Comment (henrytill) [inline execution_engine/src/core/engine_state/genesis.rs:562]:
```suggestion
```

File: execution_engine/src/core/engine_state/genesis.rs [hunk 3]
```diff
@@ -417,6 +418,831 @@ impl Distribution<ExecConfig> for Standard {
...
+            named_keys.insert(TOTAL_SUPPLY_KEY.to_string(), total_supply_uref.into());
+
```
Comment (henrytill) [inline execution_engine/src/core/engine_state/genesis.rs:564]:
```suggestion
```

File: execution_engine/src/core/engine_state/genesis.rs [hunk 4]
```diff
@@ -417,6 +418,831 @@ impl Distribution<ExecConfig> for Standard {
...
+        let mut initial_seigniorage_recipients = SeigniorageRecipientsSnapshot::new();
+        for era_id in initial_snapshot_range {
+            initial_seigniorage_recipients.insert(era_id, seigniorage_recipients.clone());
+        }
+        initial_seigniorage_recipients
+    }
+
+    fn create_purse(&self, amount: U512, deploy_hash: DeployHash) -> Result<URef, GenesisError> {
+        let tracking_copy = Rc::clone(&self.tracking_copy);
```
Comment (henrytill) [inline execution_engine/src/core/engine_state/genesis.rs:858]: I wonder if it is necessary to do this...

**Review by henrytill — APPROVED** [review]
Couple of small things/nits and one question.  

Otherwise looking good to go.

File: execution_engine/src/core/engine_state/genesis.rs (hunk 1 above)
Comment (EdHastingsCasperLabs) [reply execution_engine/src/core/engine_state/genesis.rs]: did this

File: execution_engine/src/core/engine_state/genesis.rs (hunk 2 above)
Comment (EdHastingsCasperLabs) [reply execution_engine/src/core/engine_state/genesis.rs]: did this

File: execution_engine/src/core/engine_state/genesis.rs (hunk 3 above)
Comment (EdHastingsCasperLabs) [reply execution_engine/src/core/engine_state/genesis.rs]: did this

File: execution_engine/src/core/engine_state/genesis.rs (hunk 4 above)
Comment (EdHastingsCasperLabs) [reply execution_engine/src/core/engine_state/genesis.rs]: removed this

**Pull request merged**